src/serac/physics/utilities/functional/functional.hpp

```diff
@@ -25,45 +26,89 @@
 
 namespace serac {
 
+namespace detail {
+
 /**
- * @brief a type trait used to identify the Hcurl family
+ * @brief a (poorly named) tuple of quantities used to discover the sparsity
+ * pattern associated with element and boundary element matrices.
+ *
+ * It stores information about how the entries of an element "stiffness" matrix
+ * map to the global stiffness. The operator< definition allows us to sort them
+ * lexicographically, to facilitate creating the CSR matrix graph.
  */
-template <typename T>
-struct is_hcurl {
-  /**
-   * @brief Flag denoting the type contained is Hcurl
-   */
-  static constexpr bool value = false;
+struct ElemInfo {
+  int  global_row;
+  int  global_col;
+  int  local_row;
+  int  local_col;
+  int  element_id;
+  int  sign;
+  bool on_boundary;
 };
 
+/// @brief for sorting lexicographically by {global_row, global_col}
+inline bool operator<(const ElemInfo& x, const ElemInfo& y)
```

Comment:
Missing a lot of doxygen parameter comments in this file.

Reply:
I'm not sure how to describe `x` and `y` in a way that isn't tautological, but I'll add the @param statements to these operators nonetheless